### ISSUE TITLE
[Widget Request] BasicApp

### ISSUE THREAD

**Issue by PaulHMason** [flutter/flutter#16654]
Maybe something like this already exists and I've just missed it.

What would be very useful is a simple bare bones "App" widget (no navigation, nothing) for creating single view app. Basically, something that just sets the text direction and locale etc (a stripped down version of WidgetsApp).

**Comment by rmtmckenzie** [comment]
I think you're looking for MaterialApp. If you just set the `home` property on it to whatever you want to build, you have a working app.

**Comment by PaulHMason** [comment]
Nope, I'm looking for something far simpler. It's easy enough to write something, but I feel it should be part of the included widgets. MaterialApp is really only for Material Design apps, and WidgetsApp, while simpler, still includes multi-view navigation etc.

**Comment by zoechi** [comment]
Why do you think you need a widget for that?
You can run your custom widget.

**Comment by PaulHMason** [comment]
Not that easily - you still have to set up the text direction, localization bits and set up padding based on the window padding (so you don't overlap the status bar). There's still a fair bit that WidgetsApp and MaterialApp do that's very helpful.

**Comment by rmtmckenzie** [comment]
So what's the problem with just using the parts of MaterialApp that you actually want to use? And just because you use MaterialApp you don't necessarily have to use just Material widgets - you can put whatever you want in there.

This looks pretty simple to me, even if under the hood it does a few more things than might be absolutely necessary:
```dart
import 'package:flutter/material.dart';

void main() => runApp(new MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      home: new SafeArea(
        child: new Column(
          children: <Widget>[new Text("I'm some text!")],
        ),
      ),
    );
  }
}
```

Unless you want to do it all without a stateless widget:
```dart
void main() => runApp(
      new MaterialApp(
        home: new SafeArea(
          child: new Column(
            children: <Widget>[new Text("I'm some text!")],
          ),
        ),
      ),
    );

```

If what you're worried about is the stuff it does under the hood (i.e. navigator etc), fair enough - but if you're just doing a single page app there's very little chance of your hitting performance barriers or anything anyways, so you're just prematurely optimizing at that point...

Can you give a good reason why you need something simpler to code than what I've written there (and what it might be?)

**Comment by josvos** [comment]
+1 for this request, because of the side effects (Navigator, Theme, etc.). Note that eliminating the Navigator in MaterialApp etc. does not work (see issue #18904) despite what the documentation says.

**Comment by zoechi** [comment]
What about https://docs.flutter.io/flutter/widgets/WidgetsApp-class.html ?
